test/unit/spec/PrimaryAuth_spec.js

```diff
@@ -1447,6 +1447,17 @@ function (Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, PrimaryAuthForm, Be
             expect($.ajax.calls.count()).toBe(1);
           });
       });
+      itp('undefined username does not make API call', function () {
+        return setup({ features: { securityImage: true }})
+          .then(function (test) {
+            test.setNextResponse(resSecurityImage);
+            test.form.setUsername(undefined);
+            return waitForBeaconChange(test);
+          })
+          .then(function () {
+            expect($.ajax.calls.count()).toBe(0);
```

Comment:
In this case we show the UNDEFINED_USER security image right ? Can we also assert that the beacon gets the right classname something like `expect(test.form.securityBeacon()[0].className).toMatch('new-user')`

Reply:
@nikhilvenkatraman-okta updated.